### ISSUE TITLE
Removing child elements causes crash (Android, 7.0.3)

### ISSUE THREAD

**Issue by Bindernews** [react-native-community/react-native-svg#789]
I'm running into issues when I try to remove a child from an <Svg> element. 

Platform: Android
react-native: 0.57.0
react-native-svg: 7.0.3

Here's a picture of the error:
<img src="https://user-images.githubusercontent.com/1190457/45911085-f5420700-bddc-11e8-97be-6acfcbc264d8.png" width="400" />

It appears that React is trying to manage the removal of one of the native SVG components, but you've already removed it, but I'm not too familiar with React Native development.

Here's the reproduction code. Click the button at the top to add the line, then click again to cause the crash.
```JavaScript
import React from 'react';
import { Button, Dimensions, StyleSheet, View, } from 'react-native';
import Svg, { Polyline } from 'react-native-svg';

const { width, height } = Dimensions.get('window');

export default class SVGFail extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showLine: false
    };
  }

  toggleLine = () => {
    this.setState({ showLine: !this.state.showLine });
  }

  render() {
    const { showLine } = this.state;
    return (
      <View
        style={styles.container}
      >
        <Button
          title="Toggle Line"
          onPress={this.toggleLine}
        />
        <Svg
          width={width}
          height={height}
        >
          {showLine ? (
            <Polyline
              points={'10,10 100,100 150,100 100,150'}
              fill="none"
              stroke="red"
              strokeWidth="1"
            />
          ) : (
            null
          )}
        </Svg>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: 'beige',
  },
});
```

**UPDATE:**
After some debugging, the error occurs in NativeViewHierarchyManager.java when React tries to remove the child from the SvgView, but there isn't a child in SvgView to remove. Well, SvgView overrides the `addView` method and only adds children if they are not instances of `RenderableView`, which means that somehow SvgView never ends up with children, despite displaying the content. I'm not familiar enough with the code base to be able to help more.

**UPDATE 2:**
Everything works when I wrap my polylines with a `<G>` tag, which is great, but this is still a valid bug.

**Comment by IjzerenHein** [comment]
I'm also experiencing this crash after upgrading to react-native 0.57.
@Bindernews How did you workaround this? Could you post an example on how you wrapped the Polyline with a tag to workaround this? Thanks so much!!

**Comment by ulfgebhardt** [comment]
Same crash here

**Comment by ReggaePanda** [comment]
@IjzerenHein u can use the `G` Tag from `react-native-svg`

<img width="377" alt="bildschirmfoto 2018-09-28 um 12 28 36" src="https://user-images.githubusercontent.com/2183135/46203449-12fdf780-c31a-11e8-9c16-fed842042411.png">
<img width="632" alt="bildschirmfoto 2018-09-28 um 12 28 48" src="https://user-images.githubusercontent.com/2183135/46203451-142f2480-c31a-11e8-95da-6b63d8d4b8c4.png">

**Comment by IjzerenHein** [comment]
A sweet, thanks so much @ReggaePanda! 
@mtsap, this solves our RN57 upgrade issue.

**Comment by Bindernews** [comment]
I just updated my post with the fix. Apparently when I typed `<G>` without backticks, Markdown just swallowed it and I didn't notice. Sorry for the confusion. Leaving this open until the fix is implemented.

**Comment by GetSource1234** [comment]
Got the same issue but on ios(12.0, RN 0.57.1).

**Comment by msand** [comment]
Can you try with the latest commit from the master branch, I think I have a good enough workaround for now.

**Comment by msand** [comment]
Is this still an issue?

**Comment by Bindernews** [comment]
It was fixed for me a while ago, although apparently I forgot to mention it. Derp. If it's fixed for @GetSource1234 then I think it can be closed.

**Comment by GetSource1234** [comment]
Well, I use the same workaround as @Bindernews -> wrap elements within <G>. BTW, just verified within v. 8.0.9 and still getting an issue (env. iphone 6, 12.1 ). One more thing, form me issue occurs within  AreaChart:

issue
```
const Gradient = ({ index }) => (
        <Defs key={index}>
          <LinearGradient id={'gradient'} x1={'0%'} y={'0%'} x2={'0%'} y2={'100%'}>
            <Stop offset={'0%'} stopColor={'#0a5478'} stopOpacity={0.8}/>
            <Stop offset={'100%'} stopColor={'#0a5478'} stopOpacity={0.2}/>
          </LinearGradient>
        </Defs>
    )
<AreaChart
            data={this.state.data}
            curve={ shape.curveNatural }
            contentInset={contentInset}
            svg={{ fill: 'url(#gradient)' }}
            >
            <Gradient/>
          </AreaChart>
```

works fine
```
const Gradient = ({ index }) => (
        <G>
        <Defs key={index}>
          <LinearGradient id={'gradient'} x1={'0%'} y={'0%'} x2={'0%'} y2={'100%'}>
            <Stop offset={'0%'} stopColor={'#0a5478'} stopOpacity={0.8}/>
            <Stop offset={'100%'} stopColor={'#0a5478'} stopOpacity={0.2}/>
          </LinearGradient>
        </Defs>
      </G>
    )
<AreaChart
            data={this.state.data}
            curve={ shape.curveNatural }
            contentInset={contentInset}
            svg={{ fill: 'url(#gradient)' }}
            >
            <Gradient/>
          </AreaChart>
```

**Comment by GetSource1234** [comment]
@msand yes, mentioned patch is working, at least for me.

**Comment by msand** [comment]
I've published a proper fix for iOS in v9.2.4
If anyone has any issues in either iOS or Android with the latest version, please comment and / or provide a reproduction.
